File: 0567-permutation-in-string/0567-permutation-in-string.go
func checkInclusion(s1 string, s2 string) bool {
    l1,l2 := len(s1),len(s2)
    if l1 > l2 { return false }
    var count1, count2 [26]int
    for i := range l1 {
        count1[s1[i] - 'a']++
        count2[s2[i] - 'a']++
    }
    if count1 == count2 { return true }
    for i:=l1; i<l2; i++ {
        count2[s2[i] - 'a']++
        count2[s2[i - l1] - 'a']--
        if count2 == count1 { return true }
    }
    return false
}

File: 0567-permutation-in-string/0567-permutation-in-string.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        l1,l2 = len(s1),len(s2)
        if l1 > l2: return False
        count1 = {}
        count2 = {}
        for r in range(l1): 
            count1[s1[r]] = 1 + count1.get(s1[r],0)
            count2[s2[r]] = 1 + count2.get(s2[r],0)
        if count1 == count2: return True
        for r in range(l1,l2):
            count2[s2[r]] = 1 + count2.get(s2[r],0)
            count2[s2[r-l1]] -= 1 
            if count2[s2[r-l1]] == 0: del count2[s2[r-l1]]
            if count2 == count1: return True
        return False